tuxemon/npc.py

```diff
@@ -754,6 +754,49 @@ def remove_monster(self, monster: Monster) -> None:
             self.monsters.remove(monster)
             self.set_party_status()
 
+    def evolve_monster(self, monster: Monster, evolution: str) -> None:
+        """
+        Evolve a monster from this npc's party.
+
+        Parameters:
+            monster: Monster to remove from the npc's party.
+            evolution: Monster to add to the npc's party.
+
+        """
+        if monster not in self.monsters:
+            return
+        # Store old monster data
+        old_slot = self.monsters.index(monster)
+        old_level = monster.level
+        old_current_hp = monster.current_hp
+        old_flairs = monster.flairs
+        self.monsters.remove(monster)
+
+        # TODO: implement an evolution animation
+        # Set new monster data
+        new_monster = Monster()
+        new_monster.load_from_db(evolution)
+        new_monster.set_level(old_level)
+        new_monster.current_hp = min(old_current_hp, new_monster.hp)
+        new_monster.slug = new_monster.slug
+        self.monsters.append(new_monster)
+
+        # If evolution has a flair matching, copy it
+        for new_flair in new_monster.flairs.values():
+            for old_flair in old_flairs.values():
+                if new_flair.category == old_flair.category:
+                    new_monster.flairs[new_flair.category] = old_flair
+
+        # Removing the old monster caused all monsters in front to move
+        # a slot back
+        # Bring our new monster from the end of the list to its
+        # previous position
+        for i in range(len(self.monsters) - 1, old_slot, -1):
+            self.switch_monsters(i, i - 1)
```

Comment:
You don't need to remove, append and then switch the monsters around... Just put the new one into the array by index.

Reply:
Look, this code is from evolve_monsters, I thought it was reliable as starting point, now I recognize is full of imperfections.

Reply:
Now let's solve this point